### ISSUE TITLE
How to deal with multistore header in modules

### ISSUE THREAD

**Issue by matthieu-rolland** [PrestaShop/PrestaShop#25006]
#### Is your feature request related to a problem?

While developing the multistore module example I came upon this situation:

I have a create/edit form with a shop association block, so the element being edited can be associated to any shop or shop group: should we still display the multistore header in this situation? Even if the shop context doesn't impact the form in any way...?

#### Describe the solution you'd like

One possibility would be to:

1/ keep the multistore header in creation mode and pre-check the current context shop in the shop association block.

2/ Do not display the multistore header in edition mode.

3/ We could allow module developers to hide the multistore header when they don't need it? This feature could be a helper method in their module's controller.

Ping @marionf

**Comment by marionf** [comment]
Hey @matthieu-rolland 

This is the current behavior in the back-office (it works like this for categories for example) and described in the specs:
When you add something, the shop(s) selected in the context are automatically checked in the shop association block.
But it's possible to select/unselect shops, no matter the context selected.

When you edit something, you can unselect the shop(s) previously selected and therefore disassociate the element from these shops.
And you can also, select shop(s) not previously selected and therefore associate the element with these shops.

This behavior wasn't identified as a problem during user interviews.

So, you can go for option 2/ keep the multistore header in creation mode and pre-check the current context in the shop association block.

**Comment by matthieu-rolland** [comment]
thanks for your feedback @marionf !

Yes I saw it was the same in the category page, I just thought it was strange in edition mode, but if it's good for the users I guess we shouldn't change something they are used to  👍 

About 3/, do you think it would be an interesting feature for module developers to be able to disable the multistore header on certain pages (of their module only)? For example if their module has nothing to do with multistore?

For now, it will appear all the time in their module's pages if multistore is enabled.

**Comment by marionf** [comment]
Yes, it could be an interesting feature as some modules have no interest in being compatible with multistore, such as certain front-office modules like captchas for example

**Comment by boo-code** [comment]
Items 1 and 2 were settled here by @marionf in 2021 and the behaviour is unchanged in 9.2.x, so only item 3 is still open. There are now two render paths, and only one of them is a module's.

Legacy pages, which is what a module gets: `LegacyController` renders `Admin/Layout/legacy_layout.html.twig` (and `Admin/Module/configure.html.twig` extends it), whose selector is `component('LegacyShopList')` calling `HelperShop::getRenderedShopList()`. It renders whenever multistore is used. `$this->multishop_context` only filters which entries the dropdown offers - it cannot hide it.

Migrated pages: `default_layout.html.twig` -> `component('Toolbar')` -> `component('MultistoreHeader')`. The only opt-out there is `%prestashop.controllers_all_shop_context%`, a core container parameter a module cannot contribute to, and it locks a controller to all-shops context rather than hiding the header.

Prepared as a patch on `develop` for the legacy path: an `AdminController::$display_multishop_selector` property, `true` by default so no existing page changes, honoured by an early return in `HelperShop::getRenderedShopList()`. With it off the header falls back to the same "View my store" link a single-shop install shows, so nothing is left blank. Integration test covers both directions, and deleting the guard fails exactly the opt-out case.

One design note: I did not key this off `multishop_context == Shop::CONTEXT_ALL`, which would have needed no new property. That value means "this page only operates in all-shops context", not "this page has nothing to do with multistore", and reusing it would silently drop the selector from `AdminAccessController`, `AdminShopController` and `AdminSpecificPriceRuleController`, which is a different decision from the one asked for here.

The migrated path is deliberately untouched: no module renders through `default_layout`, so an opt-out there would have no caller today.
